File: src/main/java/com/kmatrokhin/uvbot/TelegramBeanConfig.java
package com.kmatrokhin.uvbot;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.telegram.telegrambots.client.okhttp.OkHttpTelegramClient;
import org.telegram.telegrambots.meta.generics.TelegramClient;

@Configuration
public class TelegramBeanConfig {
    @Bean
    public String telegramToken() {
        return System.getenv("TELEGRAM_TOKEN");
    }

    @Bean
    public TelegramClient telegramClient() {
        return new OkHttpTelegramClient(telegramToken());
    }

    @Bean
    public ObjectMapper objectMapper() {
        return new ObjectMapper();
    }
}

File: src/main/java/com/kmatrokhin/uvbot/config/TelegramBeanConfig.kt
package com.kmatrokhin.uvbot.config

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.telegram.telegrambots.client.okhttp.OkHttpTelegramClient
import org.telegram.telegrambots.meta.generics.TelegramClient
import java.net.http.HttpClient

@Configuration
class TelegramBeanConfig {
    @Value($$"${telegram.token}")
    private lateinit var telegramToken: String

    @Bean
    fun telegramClient(): TelegramClient {
        return OkHttpTelegramClient(telegramToken)
    }

    @Bean
    fun objectMapper(): ObjectMapper {
        return ObjectMapper()
    }

    @Bean
    fun httpClient(): HttpClient? {
        return HttpClient.newHttpClient()
    }
}
